ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/Combinator/InjectTxs.hs

```diff
@@ -97,148 +94,77 @@ matchPolyTx is tx =
--- @blk@.
-newtype ListOfTxs tx xs blk = ListOfTxs [TxWithOriginal tx xs blk]
-
--- | A special telescope. This type alias is used just for making this more
--- readable.
---
--- This in the end is basically:
---
--- > TS ... (
--- >    TZ (
--- >         [(orig, tx), ...]
--- >       , f
--- >    ) ...)
---
--- So at the tip of the telescope, we have both an @f@ and a list of tuples of
--- transactions.
-type TelescopeWithTxList g f tx xs' xs  =
-   Telescope g (Product (ListOfTxs tx xs') f) xs
-
-matchPolyTxs' ::
-     All Top xs
-  => InPairs (InjectPolyTx tx) xs
-  -> [NS tx xs]
-  -> Telescope g f xs
-  -> ( [(NS tx xs, Mismatch tx f xs)]
-     , TelescopeWithTxList g f tx xs xs
-     )
-matchPolyTxs' ips txs = go ips [ hmap (TxWithOriginal x) x | x <- txs ]
-  where
-    tipFst :: All Top xs => NS (TxWithOriginal tx xs') xs -> NS tx xs'
-    tipFst = hcollapse . hmap (K . origTx)
-
-    go :: All Top xs
-       => InPairs (InjectPolyTx tx) xs
-       -> [NS (TxWithOriginal tx xs') xs]
-       -> Telescope g f xs
-       -> ( [(NS tx xs', Mismatch tx f xs)]
-          , TelescopeWithTxList g f tx xs' xs
-          )
-    go _            txs' (TZ f)   =
-      let (rejected, accepted) =
-              partitionEithers
-            $ map (\case
-                      Z x -> Right x
-                      -- The ones from later eras are invalid
-                      S x -> Left (tipFst x, MR (hmap blkTx x) f)
-                  ) txs'
-      in (rejected, TZ (Pair (ListOfTxs accepted) f))
-
-    go (PCons i is) txs' (TS g f) =
-      let (rejected, translated) =
-              partitionEithers
-            $ map (\case
-                      Z (TxWithOriginal origx x) ->
-                        case injectTxWith i x of
-                          -- The ones from this era that we cannot transport to
-                          -- the next era are invalid
-                          Nothing -> Left (origx, ML x (Telescope.tip f))
-                          Just x' -> Right $ Z (TxWithOriginal origx x')
-                      S x -> Right x
-                  ) txs'
-          (nextRejected, nextState) = go is translated f
-      in (rejected ++ map (second MS) nextRejected, TS g nextState)
-
-matchPolyTxs ::
-     SListI xs
-  => InPairs (InjectPolyTx tx) xs
-  -> [NS tx xs]
-  -> HardForkState f xs
-  -> ( [(NS tx xs, Mismatch tx (Current f) xs)]
-     , HardForkState (Product ([] :.: tx) f) xs
-     )
-matchPolyTxs is tx =
-      fmap (HardForkState . hmap distrib)
-    . matchPolyTxs' is tx
-    . getHardForkState
-  where
-    distrib :: Product (ListOfTxs tx xs) (Current f) blk
-            -> Current (Product ([] :.: tx) f) blk
-    distrib (Pair (ListOfTxs txs) Current{..}) = Current {
-          currentStart = currentStart
-        , currentState = Pair (Comp [blkTx t | t <- txs])  currentState
-        }
-
 -- | Match transaction with an 'NS', attempting to inject where possible
-matchPolyTxNS ::
+matchPolyTxTele ::
```

Comment:
This new definition is alpha-equivalent to `matchPolyTx'` above, isn't it? So I'm dubious of this diff.

Reply:
Indeed it is, didn't realize that.